### DISCUSSION TITLE
Additionnal _id in objects

### DISCUSSION THREAD

**Discussion by billouboq** [typegoose/typegoose#411] [Q&A] [answered]
Would like to know if it's normal that since I use typegoose I get an additionnal _id to all my objects in my documents.

So if I have :

```js
_id: ObjectId,
name: string,
options: {
  hasRights: string
}
```

with typegoose model it will create :

```
_id: ObjectId,
name: string,
options: {
 _id: anAdditionnalID
  hasRights: string
}
```

And I know i can put _id: false in @props but why it isn't the default for objects ?

**Comment by hasezoey** [comment, marked as answer]
because typegoose (currently, #382) needs all object to be an extra class and all classes are an "nested schema" and an schema has always `_id` enabled by default

**Comment by billouboq** [comment]
That's nice to know, thanks for your answer !
